Emmo 1.0.0 beta3 simon

jsimonclark opened the pull request:
This pull request adapts the BattINFO structure to the new structure of EMMO in version 1.0.0-beta3. There is still some work that needs to be done to get the adaptations fully implemented. But this demonstrates the approach - will implement the rest once the approach is validated. 

Major changes include:
generic-concepts: 
- create entries for ISO80000 quantities with associated PhysicalDimensions. Most ISO80000 quantities are included.
-  introduce terms from IEC60050 where needed.
- include placeholder terms for things like Axis, Number subclasses, ScalarQuantity, VectorQuantity, TensorQuantity, etc.
- these changes should be incorporated in EMMO directly.

electrochemical-quantities: 
- introduce terms from IEC60050 where needed.

electrochemistry: 
- introduce terms from IEC60050 where needed. 
- adapt the original ElectrochemicalComponent structure to the new holistic perspective in EMMO. 
- add object properties specifically for electrochemistry including hasElectrode, hasPositiveElectrode, hasNegativeElectrode, hasElectrolyte, hasElectrodeConstituent, etc.
- define some example reactions and electrode classes for electrochemical cells
- modify the semantic description of an electrochemical cell to be the combination of two electrodes and an electrolyte. 

battery-quantities:
- introduce terms from IEC60050 where needed.

battery:
- introduce terms from IEC60050 where needed.

jesper-friis commented:
Great work! 

Some initial comments:
- RatedCapacity - change altLabel to ElectropediaReference
- OperatingTemperature and its subclasses has no elucidation. They may be too specialised to go into isq, but may still be useful at the reference level
- ElectronicConductance and IonicCondutance has no elucidation
- ElectronicConductivity and IonicCondutivity has no elucidation
- ElectronicResistance and IonicResistance has no elucidation
- ElectronicResistivity and IonicResistivity has no elucidation
- ChargeCutoffVoltage has no elucidation. Is it a battery quantity?
- TheoreticalSpecificCapacity has no elucidation

eibfl-dtu commented:
Very nice Simon!

I went through the Physicalistic branch. Here are some comments:

**LithiumCobaltOxideMaterials, LithiumGraphiteMaterials**
* Why are they defined differently (under Material) than LCO, Graphite (under active material)?
* Also they are linked to CobtalOxideSubstance concepts, but not to active material
* They refer to holistic parts CobaltOxideMolecule via their inheritance with CobaltOxideSubstance, but oxides are not understood as molecules but as ionic solids
* IntercalationSubstances appear disconnected from active material


**LCO, LNO, NCM...**
* Elucidations missing
* Consider a "IonicSolidMaterial" class to link these oxides (see further below for another use of this class). 
* Consider linking to CristallineMaterial
* External identifiers, I believe PUBCHEM CIDs are the most complete.	

**ReactiveMaterial**
* Elucidation: virtually all materials undergo chemical changes. I think the sense here is to refer to materials that undergo chemical changes when used to store energy electrochemically. If this is accepted, then we could consider eliminating it and sticking to ElectrochemicalMaterial and ActiveMaterial

**ChemicalMaterial**
* What is the context for creating this class? I believe all materials are chemical

**ElectroactiveMolecularEntity**
*  I think this class could be replaced instead by two classes, ElectrochemicalReactionProduct  (below ChemicalReactionProduct) and ElectrochemicalReactionReactant (below ChemicalReactionReactant). It sounds more consistent to me. Then they can be naturally added as participants in electrochemical reactions.

**ChargeCarrier**
* I think it should not be explicitly subclass of particle. The concept of ChargeCarrier exists in the context of a charge transport process, and these processes afaik are not defined yet. Once defined, then ChargeCarrier can be defined as participant of the process, and then make subclasses  of ChargeCarrier linked to already existing EMMO classes (electrons, holes, polarons, plasmons, ions, etc).
* Note: Number of ChargeCarrier can be linked to ElectrochemicalTransportQuantity

**Salt**
* I think it is ambiguous to put Salt under Molecule. Salt might also be a crystalline ionic solid (not a molecule). Lithium salts can also be precursors to synthesize an active material (e.g. Li2CO3). I would create a "IonicSolidMaterial" subclass of "Material" (and "MolecularSolidMaterial" as well for completeness). Then put Salt as a subclass of "IonicSolidMaterial"

jsimonclark commented:
Thanks Eibar!

Regarding your points:

**LithiumCobaltOxideMaterials, LithiumGraphiteMaterials and LCO, LNO, NCM...**
Thanks. I've addressed most of these points. They were largely due to carry-overs from previous versions that hadn't been updated yet. 

The only point for discussion is the relationship between Molecules and Materials. In this example, I contend that:
(i) it is the molecule (or molecular entity) that is a participant in the electrochemical reaction, and
(ii) the molecule is a holistic part of the material. 
The thing with substances I bypassed, since it is probably unnecessary to make a substance class for each material. 

IonicSolidMaterial might be a bit problematic. I assume that Ionic refers to the nature of the bonds. Then we would also need Covalent and Metallic Materials. In principle that should be fine and could give some added value. But, the Solid is a throwing me for a bit of a loop. In EMMO, a Solid is a Continuum, along side the other phases of matter. So, how would it be if the material were melted? It would no longer be an IonicSolidMaterial. One way we could address this is to define IonicMaterial, and then say IonicSolidMaterial is equivalent with IonicMaterial and Solid, and IonicLiquidMaterial is equivalent with IonicMaterial and Liquid?

Another thing to consider is that not all the metal oxides are ionic. Some are covalent. 

**ReactiveMaterial**
This is defined by EMMO.

**ChemicalMaterial**
This is defined by EMMO.

**ElectroactiveMolecularEntity**
I was trying to use this as a way to infer what is an active material based on what electrochemical reactions are defined, but it didn't really work as I intended. I think I'll just delete it. 

**ChargeCarrier**
The description of a ChargeCarrier comes from the IEC standard, where its particle-like nature is a central part of the definition. Of course, if we really wanted to split hairs, then it's also a wave. It's a good point, though, that we need to link this better to a ChargeTransport process.

**Salt**
Fair point. It's a bit tricky because there are a couple of ways one can define a salt. It can be a substance that is formed as the result of a reaction between an acid and a base, or it can be a chemical compound assembly of positively charged cations and negatively charged anions, which results in a compound with no net electric charge. I've given it the same treatment, where we distinguish between the SaltMolecule and the SaltMaterial
